@here/harp-mapview/lib/poi/PoiRenderer.ts

```diff
@@ -373,37 +373,21 @@ export class PoiBatchRegistry {
     }
 }
 
-// keep track of the missing textures, but only warn once
-const missingTextureName: Map<string, boolean> = new Map();
-
+// Searches the given caches for the texture name or the texture itself.
 function findImageItem(
-    poiInfo: PoiInfo,
+    imageTextureName: string,
     imageCaches: MapViewImageCache[],
     imageTexture?: ImageTexture
 ): ImageItem | undefined {
-    assert(poiInfo.imageTextureName !== undefined);
-    const imageTextureName = poiInfo.imageTextureName!;
-
-    let imageItem: ImageItem | undefined;
     for (const imageCache of imageCaches) {
-        imageItem = imageTexture
+        const imageItem = imageTexture
             ? imageCache.findImageByName(imageTexture.image)
             : imageCache.findImageByName(imageTextureName);
         if (imageItem) {
-            break;
-        }
-    }
-
-    if (!imageItem) {
-        logger.error(`init: No imageItem found with name
-            '${imageTexture?.image ?? imageTextureName}'`);
-        poiInfo.isValid = false;
-        if (missingTextureName.get(imageTextureName) === undefined) {
-            missingTextureName.set(imageTextureName, true);
-            logger.error(`preparePoi: No imageTexture with name '${imageTextureName}' found`);
+            return imageItem;
         }
     }
-    return imageItem;
+    return undefined;
```

Comment:
How do we catch actual errors, i.e. an image missing from cache that should actually be there, e.g. outdated atlas or style.

Reply:
The problem is, with the current architecture / API, we don't know when this situations occurs.

The current task queue geometry creation makes the assumption that the image cache is fixed until the geometry is created, but the user can modify the cache before the task has executed.

I break that assumption and say we can create geometry for something which doesn't exist in the image cache, but the POI is invalid, and it is up to the user to make the tiles dirty and re-request them (such is the responsibility of using a custom cache).

Reply:
I was thinking about the cache used to store theme images. The user has no control over that one. In theory we should always find images referenced in the style. 
It's quite possible that at some point we use an outdated atlas we an updated style, specially now that we don't generate it automatically. Without even an error in the log, the missing resources might go unnoticed for a long time.

I think it's better to keep logging the errors. We can make them warnings instead of errors, maybe also say in the message that it might just be a transient error.